### DISCUSSION TITLE
Verify Tag

### DISCUSSION THREAD

**Discussion by sebESAPI** [fo-dicom/fo-dicom#1540] [Q&A]
Because some dicom files differ from vendor to vendor due to standard/specifications, how do I check if  DICOM Tag exist then check if it has a value.  For example,  the following code will ONLY work if the tag exists but will fail fail if the tag does not exist
var tag = Dicom.DicomTag.Parse("0010,0040");
 var result = dicomDataset.GetString(tag);
                if (result == "")
                {
                   // MessageBox.Show("The Patient's Sex is empty");
                                  
                }

                else
                {
                    MessageBox.Show("The Patient's Sex has a value");
                }

so I need to check both, existence of tag and the value

**Comment by mrbean-bremen** [comment]
You can use [TryGetString()](https://fo-dicom.github.io/stable/v5/api/FellowOakDicom.DicomDataset.html#FellowOakDicom_DicomDataset_TryGetString_FellowOakDicom_DicomTag_System_String__) to get a value from a tag you don't know if it exists:
```cs
string patientSex;
if (dataset.TryGetString(DicomTag.PatientSex, out string patientSex) && !string.isNullOrEmpty(patientSex)) 
{
    // handle sex
}
else
{
  // handle missing or empty tag
}
```
You could also use the more convenient [GetValueOrDefault()](https://fo-dicom.github.io/stable/v5/api/FellowOakDicom.DicomDataset.html#FellowOakDicom_DicomDataset_GetValueOrDefault__1_FellowOakDicom_DicomTag_System_Int32___0_) in this case, as you are not interested in the difference between a non-existing and an empty tag:
```cs
string patientSex = dataset.GetValueOrDefault(DicomTag.PatientSex, 0, "");
```
